rpcs3/util/fixed_typemap.hpp

```diff
@@ -313,7 +313,12 @@ namespace stx
 			// Destroy objects in reverse order
 			for (; _max; _max--)
 			{
-				(*--m_info)->destroy(*--m_order);
+				auto* info = *--m_info;
+				const u32 type_index = static_cast<const type_info<typeinfo>*>(info)->index();
+				info->destroy(*--m_order);
+
+				// Set init to false. We don't want other fxo to use this fxo in their destructor.
+				m_init[type_index] = false;
```

Comment:
Do it in info->destroy

Reply:
It's not the same object?

Reply:
I mean, move this line to typeinfo::destroy. It is the base class.

Reply:
it's a static function...

Reply:
Then pass it as argument or smth

Reply:
And do it in auto_typemap.hpp too

Reply:
Sorry but I don't have the time to make this code even more spaghetti-like